### DISCUSSION TITLE
JSON message via websocket for get_wallet_balance

### DISCUSSION THREAD

**Discussion by baerrs** [Chia-Network/chia-blockchain#10514] [Support]
Can someone help, I've can connect via WebSocket and get broadcast information back.  
I want to get get_wallet_balance info but don't know how to request the information.  
I'm currently sending 

```
{"destination": "wallet", "ack": True, "command": "get_wallet_balance", "request_id": "123456w", "origin": "ui", "data": {"wallet_id": "1"}}
```
When selecting a wallet via the GUI, I do get the information via WebSockets.

I'm looking for the proper JSON message to sent to the websocket.

**Comment by loppefaaret** [comment]
we have just had some changes to the wallet coming in, so the docs aren't quite there yet.
best i can offer for now is https://docs.chia.net/docs/12rpcs/rpcs

**Reply by baerrs** [reply]
Thanks ! I'll wait for an update in the docs (well, kind of.. I'm sure I'll be playing to see if I can figure it out).

**Reply by baerrs** [reply]
Well, I knew I wouldn't sit idle.  to make the daemon return the get_wallet_balance at the start of the client I needed to send this  I have no idea right now if this will give periodic updates, or if it's a one-time.  I'll need to wait till I get another chia coin win.  

UPDATE:  This is only part of the equation to get the data.  For this to work, I need to be logged in to the GUI.   This is not my intended functionality.  I will work on this more in the coming days.  There must be a "Log In" function the GUI is doing that i'm not.  Help is wanted if anyone is looking at this.  

```
    message = {"destination": "daemon", "command": "unlock_keyring",
               "request_id": "123456ca", "origin": "", "data": {"key": "<key phrase>"}}
    on_send_message(self, message)
```

This can be added to the on_open function @ https://docs.chia.net/docs/12rpcs/rpcs
If/when this is added, can someone edit the on_close function as well?  
```
def on_close(self, ws, e):
    print("{0]: Websocket closed: {1}".format(datetime.now(), e))
```
to 
```
def on_close(self, ws, e):
    print("{0}: Websocket closed: {1}".format(datetime.now(), e))
```
